common/src/main/java/org/apache/falcon/persistence/ExtensionBean.java

```diff
@@ -44,7 +44,9 @@
         @NamedQuery(name = PersistenceConstants.GET_ALL_EXTENSIONS, query = "select OBJECT(a) from ExtensionBean a "),
         @NamedQuery(name = PersistenceConstants.DELETE_EXTENSIONS_OF_TYPE, query = "delete from ExtensionBean a where a.extensionType = :extensionType "),
         @NamedQuery(name = PersistenceConstants.DELETE_EXTENSION, query = "delete from ExtensionBean a where a.extensionName = :extensionName "),
-        @NamedQuery(name = PersistenceConstants.GET_EXTENSION, query = "select OBJECT(a) from ExtensionBean a where a.extensionName = :extensionName")
+        @NamedQuery(name = PersistenceConstants.GET_EXTENSION, query = "select OBJECT(a) from ExtensionBean a where a.extensionName = :extensionName"),
+        @NamedQuery(name = PersistenceConstants.ENABLE_EXTENSION, query = "update ExtensionBean OBJECT(a) set is_enabled = true where a.extensionName = :extensionName"),
```

Comment:
Having a status in the DB is far more extensible that having a boolean. Else, it will involve DB migration later. You can have a simple enum with the ENABLED and DISABLED as statuses. And, store them in the DB.